smooth/components/component_battery.py

```diff
@@ -200,26 +222,21 @@ def prepare_simulation(self, components):
 
         # ToDo: c_rate depending on the soc
 
-        # Max. chargeable or dischargeable power [W] going in from the bus
-        # due to c_rate depending on the soc. To ensure that the battery can
-        # be fully charged in one timestep, the nominal value of the input-flow
-        # needs to be higher than what's actually going into the battery.
-        # Therefore we need to divide by the efficiency_charge.  Due to the
-        # inflow_conversion_factor (in "create oemof model") the battery will
-        # then receive right amount.
-        # Flows are calculated as Power [W], so the amount of charge- or dischargeable
-        # energy [Wh] has to be divided by the intervaltime [h]  (P = E / t)
-        # The c-rate is given in [W/Wh], multiplication with capacity [Wh] results in power [W]
-
-        self.p_in_max = min(
-            self.c_rate_charge * self.battery_capacity,
-            (self.battery_capacity - self.soc * self.battery_capacity) /
-            (self.sim_params.interval_time/60)
-            ) / self.efficiency_charge
-        self.p_out_max = min(
-            self.c_rate_discharge * self.battery_capacity,
-            (self.soc * self.battery_capacity) / (self.sim_params.interval_time/60)
-            )
+        # The maximum chargeable or dischargeable power [W] going in to or out of the battery is
+        # dependant on the C-rate and the capacity:
+        # p_charge_max = battery_capacity * c_rate_charge
+        # p_discharge_max = battery_capacity * c_rate_discharge
+        # To ensure that the battery can be charged and discharged within the time frame inherently
+        # defined by the C-rate (1C : Full capacity can be (dis-)charged within one hour), the
+        # nominal value of the input-flow/output-flow between bus and battery needs to include the
+        # energy losses during the (dis-)charging process:
+        # p_in_max = p_charge_max / efficiency_charge
+        # p_out_max = p_discharge_max * efficiency_charge
+        # Due to the inflow_conversion_factor / outflow_conversion_factor (in "create oemof model")
+        # the battery will then receive right amount.
```

Comment:
Maybe a reference to the text above would be sufficient, to avoid redundance and summerize in fewer words like:
"As explained above, the maximum in-/outflow of power are derived from the efficiency-loss compensated maximum (dis-)charging power implied by C-rate and battery capacity."